File: v1/172.py
class Solution(object):
    '''算法思路：
    观察规律可以看到，尾 0 的个数与 <= n 的数中 5 的个数相等
    '''
    def trailingZeroes(self, n):
        count, div = 0, n
        while div:
            div //= 5
            count += div
        return count


s = Solution()
print(s.trailingZeroes(18))

File: go/172.go

//package main
//
//import "fmt"

//6 star,  0是由2*5得到，2足够，所有就是求有多少个5的因子,
//n = 128, 第一遍 n/5 得到 25， 有25个5的倍数， 25／5 =5 ，有5个25的倍数， 5／5 = 1， 有1个125的倍数
//结果就是25 + 5 + 1， 注意为什么不是25 + 5 * 2 + 3 * 1, 因为前边已经算上了，所以 5 和 1 都只是加一次

func trailingZeroes(n int) int {
	num := 5
	count := 0
	for n >= num {
		count += n / num
		num *= 5
	}
	return count

}


//func main() {
//	fmt.Println(trailingZeroes(128))
//}